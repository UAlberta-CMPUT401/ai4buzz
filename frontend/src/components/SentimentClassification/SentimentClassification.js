import React from 'react';

const SentimentClassification = ({ results }) => {
  return (
    <div>
      {Object.keys(results).map((sentimentClass, idx) => {
        return (
          <div key={idx}>
            <div>{sentimentClass}</div>
            <div className="value">{results[sentimentClass]}</div>
          </div>
        );
      })}
    </div>
  );
};

export default SentimentClassification;
