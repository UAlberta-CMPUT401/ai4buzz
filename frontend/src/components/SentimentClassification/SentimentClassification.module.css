.sentimentClassificationData {
  display: flex;
  justify-content: space-between;
}
