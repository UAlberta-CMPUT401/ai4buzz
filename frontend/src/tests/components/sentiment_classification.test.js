import React from 'react';
import SentimentClassification from '../../components/SentimentClassification/SentimentClassification';
import renderer from 'react-test-renderer';

it('Renders ImageClassification component correctly', () => {
  const sentimentClassificationResults = {
    Negative: '68.18%',
    Neutral: '15.44%',
    Postive: '16.38%',
  };
  const tree = renderer
    .create(
      <SentimentClassification results={sentimentClassificationResults} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
