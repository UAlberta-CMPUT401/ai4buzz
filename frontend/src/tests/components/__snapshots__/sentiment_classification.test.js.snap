// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders ImageClassification component correctly 1`] = `
<div>
  <div>
    <div>
      Negative
    </div>
    <div
      className="value"
    >
      68.18%
    </div>
  </div>
  <div>
    <div>
      Neutral
    </div>
    <div
      className="value"
    >
      15.44%
    </div>
  </div>
  <div>
    <div>
      Postive
    </div>
    <div
      className="value"
    >
      16.38%
    </div>
  </div>
</div>
`;
